src/lib/routing/lib/slug.ts

```diff
@@ -0,0 +1,38 @@
+import type { SiteLocale } from '@lib/datocms/types';
+
+export const missingSlug = '-';
+export type MaybeSlug = string | undefined;
+
+type LocalizedSlugs = {
+  _allSlugLocales?:
+    | {
+        locale?: SiteLocale | null;
+        value?: string;
+      }[]
+    | null;
+};
```

Comment:
A lot of `null` and `undefined` in here, but it is important that these definitions look a lot like the generated types from DatoCMS.